examples/reshare/src/dkg/actor.rs

```diff
@@ -1,137 +1,148 @@
-use super::{Mailbox, Message};
+use super::{
+    state::{Dealer, Epoch as EpochState, Player, PlayerResponse, Storage},
+    Mailbox, Message as MailboxMessage, PostUpdate, Update, UpdateCallBack,
+};
 use crate::{
-    dkg::{manager::RoundResult, DealOutcome, DkgManager},
+    namespace,
     orchestrator::{self, EpochTransition},
-    setup::ParticipantConfig,
+    setup::PeerConfig,
     BLOCKS_PER_EPOCH,
 };
-use commonware_codec::{Encode, EncodeSize, RangeCfg, Read, ReadExt, Write};
+use bytes::{Buf, BufMut};
+use commonware_codec::{Encode, EncodeSize, Error as CodecError, Read, ReadExt, Write};
 use commonware_consensus::{
-    types::{Epoch, EpochDelta},
-    utils::{epoch, is_last_block_in_epoch, relative_height_in_epoch},
+    types::Epoch,
+    utils::{epoch as compute_epoch, is_last_block_in_epoch, relative_height_in_epoch},
     Reporter,
 };
 use commonware_cryptography::{
     bls12381::{
-        dkg::{player::Output, types::Ack},
-        primitives::{group::Share, poly::Public, variant::Variant},
+        dkg::{observe, DealerPrivMsg, DealerPubMsg, Info, Output, PlayerAck},
+        primitives::{group::Share, variant::Variant},
     },
-    Hasher, Signer,
+    transcript::Summary,
+    Digest as _, Hasher, PublicKey, Signer,
 };
-use commonware_p2p::{utils::mux::Muxer, Manager, Receiver, Sender};
-use commonware_runtime::{spawn_cell, Clock, ContextCell, Handle, Metrics, Spawner, Storage};
-use commonware_storage::metadata::Metadata;
-use commonware_utils::{
-    fixed_bytes, hex,
-    ordered::Set,
-    quorum,
-    sequence::{FixedBytes, U64},
-    Acknowledgement, TryCollect, NZU32,
+use commonware_macros::select;
+use commonware_p2p::{utils::mux::Muxer, Manager, Receiver, Recipients, Sender};
+use commonware_runtime::{
+    spawn_cell, Clock, ContextCell, Handle, Metrics, Spawner, Storage as RuntimeStorage,
 };
+use commonware_utils::{ordered::Set, Acknowledgement as _, NZU32};
 use futures::{channel::mpsc, StreamExt};
-use governor::{clock::Clock as GClock, Quota};
-use prometheus_client::metrics::counter::Counter;
-use rand::{
-    rngs::StdRng,
-    seq::{IteratorRandom, SliceRandom},
-    SeedableRng,
+use governor::{
+    clock::Clock as GClock, middleware::NoOpMiddleware, state::keyed::HashMapStateStore, Quota,
+    RateLimiter,
 };
 use rand_core::CryptoRngCore;
-use std::{cmp::Ordering, collections::BTreeMap, path::PathBuf};
-use tracing::info;
+use std::num::NonZeroU32;
+use tracing::{debug, info, warn};
+
+/// Rate limiter for messages.
+type RateLimit<K, E> =
```

Comment:
Non-actionable, but this type is huge and we use it in a few places in the codebase. #2414 adds an alias in `utils` that will be nice to use across the board.

Reply:
Oooo nice, yeah that'll be great